### ISSUE TITLE
Upgrade module to 0.12

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/terraform-aws-s3-cloudfront-website#28]
Some incompatibilities like:

```
Warning: Quoted references are deprecated

  on .terraform/modules/site-root/cloudfront.tf line 7, in resource "aws_cloudfront_distribution" "main":
   7:   provider = "aws.cloudfront"

In this context, references are expected literally rather than in quotes.
Terraform 0.11 and earlier required quotes, but quoted references are now
deprecated and will be removed in a future version of Terraform. Remove the
quotes surrounding this reference to silence this warning.

(and 3 more similar warnings elsewhere)


Warning: Interpolation-only expressions are deprecated

  on .terraform/modules/site-root/cloudfront.tf line 12, in resource "aws_cloudfront_distribution" "main":
  12:     domain_name = "${aws_s3_bucket.main.website_endpoint}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and 8 more similar warnings elsewhere)


Warning: Quoted type constraints are deprecated

  on .terraform/modules/site-root/variables.tf line 2, in variable "fqdn":
   2:   type        = "string"

Terraform 0.11 and earlier required type constraints to be given in quotes,
but that form is now deprecated and will be removed in a future version of
Terraform. To silence this warning, remove the quotes around "string".

(and 6 more similar warnings elsewhere)
```

@phuonghuynh can you help fix them? Thanks!

**Comment by ronaldtse** [comment]
@phuonghuynh please also upgrade the other terraform-aws-* modules we have using the same procedure (tag for 0.11, release for 0.12). Thanks!

**Comment by ronaldtse** [comment]
Ping @phuonghuynh .

**Comment by phuonghuynh** [comment]
Latest version is ready in v12, https://github.com/riboseinc/terraform-aws-s3-cloudfront-website/releases/tag/1.0.2

**Comment by ronaldtse** [comment]
But I still see these issues?

```
Warning: Quoted references are deprecated

  on .terraform/modules/site-main/cloudfront.tf line 7, in resource "aws_cloudfront_distribution" "main":
   7:   provider = "aws.cloudfront"

In this context, references are expected literally rather than in quotes.
Terraform 0.11 and earlier required quotes, but quoted references are now
deprecated and will be removed in a future version of Terraform. Remove the
quotes surrounding this reference to silence this warning.

(and 7 more similar warnings elsewhere)


Warning: Interpolation-only expressions are deprecated

  on .terraform/modules/site-main/cloudfront.tf line 12, in resource "aws_cloudfront_distribution" "main":
  12:     domain_name = "${aws_s3_bucket.main.website_endpoint}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and 17 more similar warnings elsewhere)


Warning: Quoted type constraints are deprecated

  on .terraform/modules/site-main/variables.tf line 2, in variable "fqdn":
   2:   type        = "string"

Terraform 0.11 and earlier required type constraints to be given in quotes,
but that form is now deprecated and will be removed in a future version of
Terraform. To silence this warning, remove the quotes around "string".

(and 13 more similar warnings elsewhere)


------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.

```

**Comment by ronaldtse** [comment]
Already addressed in c17aee4582a88105109ed51fcccf6c0ef783096d.
